### ISSUE TITLE
go get fails

### ISSUE THREAD

**Issue by vvidic** [ClusterLabs/hawk-apiserver#15]
Maybe some problems with the submodule config?

```
$ go get github.com/ClusterLabs/hawk-apiserver/
# cd /home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver; git submodule update --init --recursive
Submodule 'vendor/github.com/krig/go-pacemaker' (git@github.com:krig/go-pacemaker) registered for path 'vendor/github.com/krig/go-pacemaker'
Submodule 'vendor/github.com/sirupsen/logrus' (git@github.com:sirupsen/logrus) registered for path 'vendor/github.com/sirupsen/logrus'
Submodule 'vendor/golang.org/x/crypto' (git@github.com:golang/crypto) registered for path 'vendor/golang.org/x/crypto'
Submodule 'vendor/golang.org/x/net' (git@github.com:golang/net) registered for path 'vendor/golang.org/x/net'
Submodule 'vendor/golang.org/x/sys' (git@github.com:golang/sys) registered for path 'vendor/golang.org/x/sys'
Submodule 'vendor/gopkg.in/xmlpath.v2' (git@github.com:go-xmlpath/xmlpath.git) registered for path 'vendor/gopkg.in/xmlpath.v2'
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/krig/go-pacemaker'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:krig/go-pacemaker' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/krig/go-pacemaker' failed
Failed to clone 'vendor/github.com/krig/go-pacemaker'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/sirupsen/logrus'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:sirupsen/logrus' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/sirupsen/logrus' failed
Failed to clone 'vendor/github.com/sirupsen/logrus'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/sys'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/sys' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/sys' failed
Failed to clone 'vendor/golang.org/x/sys'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/crypto'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/crypto' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/crypto' failed
Failed to clone 'vendor/golang.org/x/crypto'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/net'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/net' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/net' failed
Failed to clone 'vendor/golang.org/x/net'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/net'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/net' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/net' failed
Failed to clone 'vendor/golang.org/x/net' a second time, aborting
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/gopkg.in/xmlpath.v2'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:go-xmlpath/xmlpath.git' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/gopkg.in/xmlpath.v2' failed
Failed to clone 'vendor/gopkg.in/xmlpath.v2'. Retry scheduled
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/sirupsen/logrus'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:sirupsen/logrus' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/sirupsen/logrus' failed
Failed to clone 'vendor/github.com/sirupsen/logrus' a second time, aborting
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/krig/go-pacemaker'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:krig/go-pacemaker' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/github.com/krig/go-pacemaker' failed
Failed to clone 'vendor/github.com/krig/go-pacemaker' a second time, aborting
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/sys'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/sys' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/sys' failed
Failed to clone 'vendor/golang.org/x/sys' a second time, aborting
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/crypto'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:golang/crypto' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/golang.org/x/crypto' failed
Failed to clone 'vendor/golang.org/x/crypto' a second time, aborting
Cloning into '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/gopkg.in/xmlpath.v2'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:go-xmlpath/xmlpath.git' into submodule path '/home/vvidic/go/src/github.com/ClusterLabs/hawk-apiserver/vendor/gopkg.in/xmlpath.v2' failed
Failed to clone 'vendor/gopkg.in/xmlpath.v2' a second time, aborting
package github.com/ClusterLabs/hawk-apiserver: exit status 1
```

**Comment by liangxin1300** [comment]
Personally I think you should do this firstly: https://help.github.com/articles/connecting-to-github-with-ssh/

**Comment by vvidic** [comment]
On Thu, Feb 14, 2019 at 11:48:20PM -0800, Xin Liang wrote:
> Personally I think you should do this firstly: https://help.github.com/articles/connecting-to-github-with-ssh/

Could be, but many other go modules from github work without
ssh setup so this behaviour is a bit strange IMHO.

-- 
Valentin

**Comment by krig** [comment]
Yeah, that's because the modules are vendored using the github SSH syntax; `git@github.com...`

I'm somewhat new to go and the module/vendoring best practices keep changing, so things are probably not set up in the best way. I'll take a look at other projects to see if I can figure out how to do this right.

**Comment by krig** [comment]
Should be fixed by #18
